kie-server-parent/kie-server-services/kie-server-services-jbpm/src/main/java/org/kie/server/services/jbpm/JbpmKieServerExtension.java

```diff
@@ -708,12 +709,19 @@ protected void addAsyncHandler(final KModuleDeploymentUnit unit, final InternalK
     protected void addTaskBAMEventListener(final KModuleDeploymentUnit unit, final InternalKieContainer kieContainer) {
         final DeploymentDescriptor descriptor = getDeploymentDescriptor(unit, kieContainer);
         if (descriptor.getAuditMode() != AuditMode.NONE) {
-            descriptor.getBuilder().addTaskEventListener(
-                    new ObjectModel(
+            List<ObjectModel> oldList = descriptor.getTaskEventListeners();
+            ObjectModel model = new ObjectModel(
                             "mvel",
                             "new org.jbpm.services.task.lifecycle.listeners.BAMTaskEventListener(false)"
-                    )
             );
+            if(!oldList.contains(model)) {
+                if(!oldList.isEmpty()) {
+                    oldList.add(0, model);
+                } else {
+                    oldList.add(model);
```

Comment:
even if the list is empty, add(0, model) would add it in the first position, is this if-else worth it?

Reply:
done